src/ray/object_manager/push_manager.cc

```diff
@@ -33,13 +33,21 @@ void PushManager::StartPush(const NodeID &dest_id, const ObjectID &obj_id,
   ScheduleRemainingPushes();
 }
 
-void PushManager::OnChunkComplete(const NodeID &dest_id, const ObjectID &obj_id) {
+void PushManager::OnChunkComplete(const ray::Status &status, int64_t chunk_id,
+                                  const NodeID &dest_id, const ObjectID &obj_id) {
   auto push_id = std::make_pair(dest_id, obj_id);
-  chunks_in_flight_ -= 1;
-  if (--push_info_[push_id]->chunks_remaining <= 0) {
-    push_info_.erase(push_id);
-    RAY_LOG(DEBUG) << "Push for " << push_id.first << ", " << push_id.second
-                   << " completed, remaining: " << NumPushesInFlight();
+  --chunks_in_flight_;
+  if (status.ok()) {
+    if (--push_info_[push_id]->chunks_remaining <= 0) {
+      push_info_.erase(push_id);
+      RAY_LOG(DEBUG) << "Push for " << dest_id << ", " << obj_id
+                     << " completed, remaining: " << NumPushesInFlight();
+    }
+  } else {
```

Comment:
I plan only resend error due to:  `status.IsOutOfMemory() || status.IsTimedOut() || status.IsObjectStoreFull() || status.IsTransientObjectStoreFull())`, but it looks like we convert all `grpc::Status` to `ray::Status::IOError` . Extra information is needed to be returned.